### PR TITLE
Some fixes

### LINKED ISSUE #19: address memory leaks by using smart pointers
xref https://github.com/desktop/registry-js/pull/9#discussion_r154713480

cc @danilaml

### PR DESCRIPTION
**Pull request by danilaml**
fixes #19

Also, it looks like `ReadValues` function is wrong (double is casted to `ulong` there and then to `HKEY`).

### CODE REVIEW

File: test/registry-test.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ if (process.platform === 'win32') {
 
       const programFilesDir = values.find(v => v.name == 'ProgramFilesDir')
       expect(programFilesDir!.type).equals('REG_SZ')
-      expect(programFilesDir!.data).contains('C:\\Program Files')
+      expect(programFilesDir!.data).contains(':\\Program Files')
```
Comment (shiftkey) [inline test/registry-test.ts]: This change is because you've got Windows installed on a different drive, right?

Comment (danilaml) [reply test/registry-test.ts]: @shiftkey yep.

**Comment by shiftkey** [comment]
> Also, it looks like `ReadValues` function is wrong (double is casted to `ulong` there and then to HKEY).

Yeah, I left those warnings around to remind me to come back to it:



I'll open a proper issue to track it.

**Comment by shiftkey** [comment]
See https://github.com/desktop/registry-js/issues/21 for those compiler warnings

**Pull request merged**